airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessageFactory.kt

```diff
@@ -207,10 +206,20 @@ class DestinationMessageFactory(
         val descriptor =
             namespaceMapper.map(
                 namespace = streamState.streamDescriptor.namespace,
-                name = streamState.streamDescriptor.name
+                name = streamState.streamDescriptor.name,
             )
+
+        val (unmappedNamespace: String?, unmappedName: String) =
+            try {
+                val stream = catalog.getStream(descriptor)
+                stream.unmappedNamespace to stream.unmappedName
+            } catch (e: Exception) {
```

Comment:
what exception is this catching? (if it's the catalog.getStream lookup, would it be better to just blow up loudly?)

Reply:
can you add that as a comment in the code?

(long term I feel like we should try to stop parsing global states' stream states at all :/ not a blocker for your PR though)